<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference • SpatialDDLS</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Function reference"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialDDLS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href=".././articles/realModelExample-hq.html">Get started! Deconvolution of mouse lymph node samples</a>
    <a class="dropdown-item" href=".././articles/hdf5Backend.html">HDF5 files</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Functions</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/diegommcc/SpatialDDLS">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="DeconvDLModel-class.html">DeconvDLModel-class</a></code> <code><a href="DeconvDLModel-class.html">DeconvDLModel</a></code> 
        </dt>
        <dd>The DeconvDLModel Class</dd>
      </dl><dl><dt>
          
          <code><a href="PropCellTypes-class.html">PropCellTypes-class</a></code> <code><a href="PropCellTypes-class.html">PropCellTypes</a></code> 
        </dt>
        <dd>The PropCellTypes Class</dd>
      </dl><dl><dt>
          
          <code><a href="SpatialDDLS-Rpackage.html">SpatialDDLS-Rpackage</a></code> 
        </dt>
        <dd>SpatialDDLS: an R package to deconvolute spatial transcriptomics data using
deep neural networks</dd>
      </dl><dl><dt>
          
          <code><a href="SpatialDDLS-class.html">SpatialDDLS-class</a></code> <code><a href="SpatialDDLS-class.html">SpatialDDLS</a></code> 
        </dt>
        <dd>The SpatialDDLS Class</dd>
      </dl><dl><dt>
          
          <code><a href="ZinbParametersModel-class.html">ZinbParametersModel-class</a></code> <code><a href="ZinbParametersModel-class.html">ZinbParametersModel</a></code> 
        </dt>
        <dd>The Class ZinbParametersModel</dd>
      </dl><dl><dt>
          
          <code><a href="barErrorPlot.html">barErrorPlot()</a></code> 
        </dt>
        <dd>Generate bar error plots</dd>
      </dl><dl><dt>
          
          <code><a href="barPlotCellTypes.html">barPlotCellTypes()</a></code> 
        </dt>
        <dd>Bar plot of deconvoluted cell type proportions</dd>
      </dl><dl><dt>
          
          <code><a href="blandAltmanLehPlot.html">blandAltmanLehPlot()</a></code> 
        </dt>
        <dd>Generate Bland-Altman agreement plots between predicted and expected cell
type proportions of test data</dd>
      </dl><dl><dt>
          
          <code><a href="calculateEvalMetrics.html">calculateEvalMetrics()</a></code> 
        </dt>
        <dd>Calculate evaluation metrics on test mixed transcriptional profiles</dd>
      </dl><dl><dt>
          
          <code><a href="cell.names.html">cell.names()</a></code> <code><a href="cell.names.html">`cell.names&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>cell.names</code> slot in a
<code>PropCellTypes</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="cell.types.html">cell.types()</a></code> <code><a href="cell.types.html">`cell.types&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>cell.types</code> slot in a
<code>DeconvDLModel</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="corrExpPredPlot.html">corrExpPredPlot()</a></code> 
        </dt>
        <dd>Generate correlation plots between predicted and expected cell type
proportions of test data</dd>
      </dl><dl><dt>
          
          <code><a href="createSpatialDDLSobject.html">createSpatialDDLSobject()</a></code> 
        </dt>
        <dd>Create a <code>SpatialDDLS</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="deconv.spots.html">deconv.spots()</a></code> <code><a href="deconv.spots.html">`deconv.spots&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>deconv.spots</code> slot in a
<code>SpatialDDLS</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="deconvSpatialDDLS.html">deconvSpatialDDLS()</a></code> 
        </dt>
        <dd>Deconvolute spatial transcriptomics data using trained model</dd>
      </dl><dl><dt>
          
          <code><a href="distErrorPlot.html">distErrorPlot()</a></code> 
        </dt>
        <dd>Generate box or violin plots showing error distribution</dd>
      </dl><dl><dt>
          
          <code><a href="estimateZinbwaveParams.html">estimateZinbwaveParams()</a></code> 
        </dt>
        <dd>Estimate parameters of the ZINB-WaVE model to simulate new single-cell
RNA-Seq expression profiles</dd>
      </dl><dl><dt>
          
          <code><a href="features.html">features()</a></code> <code><a href="features.html">`features&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>features</code> slot in a
<code>DeconvDLModel</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="genMixedCellProp.html">genMixedCellProp()</a></code> 
        </dt>
        <dd>Generate training and test cell type composition matrices</dd>
      </dl><dl><dt>
          
          <code><a href="getProbMatrix.html">getProbMatrix()</a></code> 
        </dt>
        <dd>Getter function for the cell composition matrix</dd>
      </dl><dl><dt>
          
          <code><a href="installTFpython.html">installTFpython()</a></code> 
        </dt>
        <dd>Install Python dependencies for SpatialDDLS</dd>
      </dl><dl><dt>
          
          <code><a href="interGradientsDL.html">interGradientsDL()</a></code> 
        </dt>
        <dd>Calculate gradients of predicted cell types/loss function with respect to
input features for interpreting trained deconvolution models</dd>
      </dl><dl><dt>
          
          <code><a href="loadSTProfiles.html">loadSTProfiles()</a></code> 
        </dt>
        <dd>Loads spatial transcriptomics data into a SpatialDDLS object</dd>
      </dl><dl><dt>
          
          <code><a href="loadTrainedModelFromH5.html">loadTrainedModelFromH5()</a></code> 
        </dt>
        <dd>Load from an HDF5 file a trained deep neural network model into a
<code>SpatialDDLS</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="method.html">method()</a></code> <code><a href="method.html">`method&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>method</code> slot in a <code>PropCellTypes</code>
object</dd>
      </dl><dl><dt>
          
          <code><a href="mixed.profiles.html">mixed.profiles()</a></code> <code><a href="mixed.profiles.html">`mixed.profiles&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>mixed.profiles</code> slot in a <code>SpatialDDLS</code>
object</dd>
      </dl><dl><dt>
          
          <code><a href="model.html">model()</a></code> <code><a href="model.html">`model&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>model</code> slot in a <code>DeconvDLModel</code>
object</dd>
      </dl><dl><dt>
          
          <code><a href="plotDistances.html">plotDistances()</a></code> 
        </dt>
        <dd>Plot distances between intrinsic and extrinsic profiles</dd>
      </dl><dl><dt>
          
          <code><a href="plotHeatmapGradsAgg.html">plotHeatmapGradsAgg()</a></code> 
        </dt>
        <dd>Plot a heatmap of gradients of classes / loss function wtih respect to the
input</dd>
      </dl><dl><dt>
          
          <code><a href="plotSpatialClustering.html">plotSpatialClustering()</a></code> 
        </dt>
        <dd>Plot results of clustering based on predicted cell proportions</dd>
      </dl><dl><dt>
          
          <code><a href="plotSpatialGeneExpr.html">plotSpatialGeneExpr()</a></code> 
        </dt>
        <dd>Plot normalized gene expression data (logCPM) in spatial coordinates</dd>
      </dl><dl><dt>
          
          <code><a href="plotSpatialProp.html">plotSpatialProp()</a></code> 
        </dt>
        <dd>Plot predicted proportions for a specific cell type using spatial coordinates
of spots</dd>
      </dl><dl><dt>
          
          <code><a href="plotSpatialPropAll.html">plotSpatialPropAll()</a></code> 
        </dt>
        <dd>Plot predicted proportions for all cell types using spatial coordinates of
spots</dd>
      </dl><dl><dt>
          
          <code><a href="plotTrainingHistory.html">plotTrainingHistory()</a></code> 
        </dt>
        <dd>Plot training history of a trained SpatialDDLS deep neural network model</dd>
      </dl><dl><dt>
          
          <code><a href="plots.html">plots()</a></code> <code><a href="plots.html">`plots&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>plots</code> slot in a <code>PropCellTypes</code>
object</dd>
      </dl><dl><dt>
          
          <code><a href="preparingToSave.html">preparingToSave()</a></code> 
        </dt>
        <dd>Prepare <code>SpatialDDLS</code> object to be saved as an RDA file</dd>
      </dl><dl><dt>
          
          <code><a href="prob.cell.types.html">prob.cell.types()</a></code> <code><a href="prob.cell.types.html">`prob.cell.types&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>prob.cell.types</code> slot in a
<code>SpatialDDLS</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="prob.matrix.html">prob.matrix()</a></code> <code><a href="prob.matrix.html">`prob.matrix&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>prob.matrix</code> slot in a
<code>PropCellTypes</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="project.html">project()</a></code> <code><a href="project.html">`project&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>project</code> slot in a <code>SpatialDDLS</code>
object</dd>
      </dl><dl><dt>
          
          <code><a href="saveRDS.html">saveRDS()</a></code> 
        </dt>
        <dd>Save <code>SpatialDDLS</code> objects as RDS files</dd>
      </dl><dl><dt>
          
          <code><a href="saveTrainedModelAsH5.html">saveTrainedModelAsH5()</a></code> 
        </dt>
        <dd>Save a trained <code>SpatialDDLS</code> deep neural network model to
disk as an HDF5 file</dd>
      </dl><dl><dt>
          
          <code><a href="set.html">set()</a></code> <code><a href="set.html">`set&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>set</code> slot in a <code>PropCellTypes</code>
object</dd>
      </dl><dl><dt>
          
          <code><a href="set.list.html">set.list()</a></code> <code><a href="set.list.html">`set.list&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>set.list</code> slot in a
<code>PropCellTypes</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="showProbPlot.html">showProbPlot()</a></code> 
        </dt>
        <dd>Show distribution plots of the cell proportions generated by
<code>genMixedCellProp</code></dd>
      </dl><dl><dt>
          
          <code><a href="simMixedProfiles.html">simMixedProfiles()</a></code> 
        </dt>
        <dd>Simulate training and test mixed spot profiles</dd>
      </dl><dl><dt>
          
          <code><a href="simSCProfiles.html">simSCProfiles()</a></code> 
        </dt>
        <dd>Simulate new single-cell RNA-Seq expression profiles using the ZINB-WaVE
model parameters</dd>
      </dl><dl><dt>
          
          <code><a href="single.cell.real.html">single.cell.real()</a></code> <code><a href="single.cell.real.html">`single.cell.real&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>single.cell.real</code> slot in a
<code>SpatialDDLS</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="single.cell.simul.html">single.cell.simul()</a></code> <code><a href="single.cell.simul.html">`single.cell.simul&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>single.cell.simul</code> slot in a
<code>SpatialDDLS</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="spatial.experiments.html">spatial.experiments()</a></code> <code><a href="spatial.experiments.html">`spatial.experiments&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>spatial.experiments</code> slot in a
<code>SpatialDDLS</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="spatialPropClustering.html">spatialPropClustering()</a></code> 
        </dt>
        <dd>Cluster spatial data based on predicted cell proportions</dd>
      </dl><dl><dt>
          
          <code><a href="test.deconv.metrics.html">test.deconv.metrics()</a></code> <code><a href="test.deconv.metrics.html">`test.deconv.metrics&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>test.deconv.metrics</code> slot in a
<code>DeconvDLModel</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="test.metrics.html">test.metrics()</a></code> <code><a href="test.metrics.html">`test.metrics&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>test.metrics</code> slot in a
<code>DeconvDLModel</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="test.pred.html">test.pred()</a></code> <code><a href="test.pred.html">`test.pred&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>test.pred</code> slot in a
<code>DeconvDLModel</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="topGradientsCellType.html">topGradientsCellType()</a></code> 
        </dt>
        <dd>Get top genes with largest/smallest gradients per cell type</dd>
      </dl><dl><dt>
          
          <code><a href="trainDeconvModel.html">trainDeconvModel()</a></code> 
        </dt>
        <dd>Train deconvolution model for spatial transcriptomics data</dd>
      </dl><dl><dt>
          
          <code><a href="trained.model.html">trained.model()</a></code> <code><a href="trained.model.html">`trained.model&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>trained.model</code> slot in a
<code>SpatialDDLS</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="training.history.html">training.history()</a></code> <code><a href="training.history.html">`training.history&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>training.history</code> slot in a
<code>DeconvDLModel</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="zinb.params.html">zinb.params()</a></code> <code><a href="zinb.params.html">`zinb.params&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>zinb.params</code> slot in a
<code>SpatialDDLS</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="zinbwave.model.html">zinbwave.model()</a></code> <code><a href="zinbwave.model.html">`zinbwave.model&lt;-`()</a></code> 
        </dt>
        <dd>Get and set <code>zinbwave.model</code> slot in a
<code>ZinbParametersModel</code> object</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://github.com/diegommcc" class="external-link">Diego Mañanes</a>, <a href="https://www.cnic.es/en/carlos-torroja" class="external-link">Carlos Torroja</a>, <a href="https://www.cnic.es/en/fatima-sanchez-cabo" class="external-link">Fatima Sanchez-Cabo</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

